src/TermInterface.jl

```diff
@@ -31,46 +54,32 @@ on `x` and must return a Symbol.
 issym(x) = false
 export issym
 
-"""
-  exprhead(x)
 
-If `x` is a term as defined by `istree(x)`, `exprhead(x)` must return a symbol,
-corresponding to the head of the `Expr` most similar to the term `x`.
-If `x` represents a function call, for example, the `exprhead` is `:call`.
-If `x` represents an indexing operation, such as `arr[i]`, then `exprhead` is `:ref`.
-Note that `exprhead` is different from `operation` and both functions should 
-be defined correctly in order to let other packages provide code generation 
-and pattern matching features. 
 """
-function exprhead end
-export exprhead
-
+  head(x)
 
+If `x` is a term as defined by `istree(x)`, `head(x)` returns the head of the
+term. If `x` represents a function call term like `f(a,b)`, the head
+is the function being called, `f`.
```

Comment:
I thought we wanted to avoid the concept of `:call` as "exprhead". 

In my latest commit `head` is the old `operation`, and `children` is the old `arguments`.

If we add both `head,children` and `operation,arguments` Metatheory.jl would just rely on `operation` and `arguments` for pattern matching. I guess the same for SU.

But what about SymbolicUtils terms? `t = f(a,b)` in SU would have `operation(t) == f`, `arguments(t) == [a,b]`, `children(t) = [f,a,b]`,  what about `head(t)`?  Should it be `SUHead()`? 
 
 I kinda dislike the idea that the users should define a struct to define the `head` of an AST node, all the information required to inspect, manipulate and create new terms is already contained in the type of the term.